### PR TITLE
chore(entities): add context-aware methods for entity tag management

### PR DESCRIPTION
**Pull request by ctrombley**

### CODE REVIEW

File: pkg/entities/tags.go [hunk 1]
```diff
@@ -24,29 +25,41 @@ type TagValue struct {
...
+// ListTags returns a collection of mutable tags for a given entity by entity GUID.
 //
 // Deprecated: Use GetTagsForEntity instead.
 func (e *Entities) ListTags(guid EntityGUID) ([]*Tag, error) {
+	return e.ListTagsWithContext(context.Background(), guid)
+}
+
+// ListTagsWithContext returns a collection of mutable tags for a given entity by entity GUID.
+//
+// Deprecated: Use GetTagsForEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:55]:
```suggestion
// Deprecated: Use GetTagsForEntityWithContext instead.
```

File: pkg/entities/tags.go [hunk 2]
```diff
@@ -58,12 +71,20 @@ func (e *Entities) ListTags(guid EntityGUID) ([]*Tag, error) {
 //
 // Deprecated: Use GetTagsForEntity instead.
 func (e *Entities) ListAllTags(guid EntityGUID) ([]*Tag, error) {
+	return e.ListAllTagsWithContext(context.Background(), guid)
+}
+
+// ListAllTagsWithContext returns a collection of all tags (mutable and not) for a given
+// entity by entity GUID.
+//
+// Deprecated: Use GetTagsForEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:80]:
```suggestion
// Deprecated: Use GetTagsForEntityWithContext instead.
```

File: pkg/entities/tags.go [hunk 3]
```diff
@@ -103,13 +124,20 @@ func filterMutable(resp listTagsResponse) ([]*Tag, error) {
 //
 // Deprecated: Use TaggingAddTagsToEntity instead.
 func (e *Entities) AddTags(guid EntityGUID, tags []Tag) error {
+	return e.AddTagsWithContext(context.Background(), guid, tags)
+}
+
+// AddTagsWithContext writes tags to the entity specified by the provided entity GUID.
+//
+// Deprecated: Use TaggingAddTagsToEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:132]:
```suggestion
// Deprecated: Use TaggingAddTagsToEntityWithContext instead.
```

File: pkg/entities/tags.go [hunk 4]
```diff
@@ -124,13 +152,20 @@ func (e *Entities) AddTags(guid EntityGUID, tags []Tag) error {
 //
 // Deprecated: Use TaggingReplaceTagsOnEntity instead.
 func (e *Entities) ReplaceTags(guid EntityGUID, tags []Tag) error {
+	return e.ReplaceTagsWithContext(context.Background(), guid, tags)
+}
+
+// ReplaceTagsWithContext replaces the entity's entire set of tags with the provided tag set.
+//
+// Deprecated: Use TaggingReplaceTagsOnEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:160]:
```suggestion
// Deprecated: Use TaggingReplaceTagsOnEntityWithContext instead.
```

File: pkg/entities/tags.go [hunk 5]
```diff
@@ -145,13 +180,20 @@ func (e *Entities) ReplaceTags(guid EntityGUID, tags []Tag) error {
 //
 // Deprecated: Use TaggingDeleteTagFromEntity instead.
 func (e *Entities) DeleteTags(guid EntityGUID, tagKeys []string) error {
+	return e.DeleteTagsWithContext(context.Background(), guid, tagKeys)
+}
+
+// DeleteTagsWithContext deletes specific tag keys from the entity.
+//
+// Deprecated: Use TaggingDeleteTagFromEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:188]:
```suggestion
// Deprecated: Use TaggingDeleteTagFromEntityWithContext instead.
```

File: pkg/entities/tags.go [hunk 6]
```diff
@@ -166,13 +208,20 @@ func (e *Entities) DeleteTags(guid EntityGUID, tagKeys []string) error {
 //
 // Deprecated: Use TaggingDeleteTagValuesFromEntity instead.
 func (e *Entities) DeleteTagValues(guid EntityGUID, tagValues []TagValue) error {
+	return e.DeleteTagValuesWithContext(context.Background(), guid, tagValues)
+}
+
+// DeleteTagValuesWithContext deletes specific tag key and value pairs from the entity.
+//
+// Deprecated: Use TaggingDeleteTagValuesFromEntity instead.
```
Comment (jbeveland27) [inline pkg/entities/tags.go:216]:
```suggestion
// Deprecated: Use TaggingDeleteTagValuesFromEntityWithContext instead.
```

**Review by jbeveland27 — COMMENTED** [review]
Just a few small tweaks to the deprecation comments

File: pkg/entities/tags.go (hunk 1 above)
Comment (ctrombley) [reply pkg/entities/tags.go:55]: 👍

File: pkg/entities/tags.go (hunk 2 above)
Comment (ctrombley) [reply pkg/entities/tags.go:80]: 👍

File: pkg/entities/tags.go (hunk 3 above)
Comment (ctrombley) [reply pkg/entities/tags.go:132]: 👍

File: pkg/entities/tags.go (hunk 4 above)
Comment (ctrombley) [reply pkg/entities/tags.go:160]: 👍

File: pkg/entities/tags.go (hunk 5 above)
Comment (ctrombley) [reply pkg/entities/tags.go:188]: 👍

File: pkg/entities/tags.go (hunk 6 above)
Comment (ctrombley) [reply pkg/entities/tags.go:216]: 👍

**Review by jbeveland27 — APPROVED** [review]
🏆

**Pull request merged**